src/components/EditDeleteBar/EditDeleteBar.test.jsx

```diff
@@ -0,0 +1,88 @@
+import { fireEvent, queryByRole, render, screen } from '@testing-library/react';
+import EditDeleteBar from './EditDeleteBar';
+
+describe('<EditDeleteBar/>', () => {
+  it('should render the edit and remove buttons when given a remove logic', () => {
+    const isVisible = true;
+    const toggleEditing = jest.fn();
+    const handleRemove = jest.fn();
+
+    const { getByRole } = render(
+      <EditDeleteBar
+        isVisible={isVisible}
+        toggleEditing={toggleEditing}
+        handleRemove={handleRemove}
+      />);
+
+    expect(getByRole('button', { name: /delete/i })).toBeVisible();
+    expect(getByRole('button', { name: /edit/i })).toBeVisible();
+  });
+
+  it('should render the edit button but not the remove one when not given a remove logic', () => {
+    const isVisible = true;
+    const toggleEditing = jest.fn();
+    const handleRemove = undefined;
+
+    const { getByRole, queryByRole } = render(
+      <EditDeleteBar
+        isVisible={isVisible}
+        toggleEditing={toggleEditing}
+        handleRemove={handleRemove}
+      />);
+
+    expect(queryByRole('button', { name: /delete/i })).toBeNull();
+    expect(getByRole('button', { name: /edit/i })).toBeVisible();
+  });
+
+  it('should not render the edit nor the remove button when visibility is set to false', () => {
+    const isVisible = false;
+    const toggleEditing = jest.fn();
+    const handleRemove = jest.fn();
+
+    const { queryByRole } = render(
+      <EditDeleteBar
+        isVisible={isVisible}
+        toggleEditing={toggleEditing}
+        handleRemove={handleRemove}
+      />);
+
+    expect(queryByRole('button', { name: /delete/i })).toBeNull();
+    expect(queryByRole('button', { name: /edit/i })).toBeNull();
+  });
+
+  it('should fire the remove event when clicking the remove button', async () => {
+    const isVisible = true;
+    const toggleEditing = jest.fn();
+    const handleRemove = jest.fn();
+
+    const { getByRole } = render(
+      <EditDeleteBar
+        isVisible={isVisible}
+        toggleEditing={toggleEditing}
+        handleRemove={handleRemove}
+      />);
+
+    const removeButton = getByRole('button', { name: /delete/i });
+    fireEvent.click(removeButton);
```

Comment:
We use `fireEvent` in some of our tests, but I think `userEvent` is a better choice. Not that it affects this test, but [this](https://stackoverflow.com/a/64008376/4838227) is handy.

Reply:
Good info, didn't know about userEvent. It's indeed a better choice. Thanks!